--- pex/compiler.py	2016-05-28 12:05:04.099343000 +0100
+++ pex/compiler.py.new	2016-05-28 15:42:36.827820391 +0100
@@ -32,6 +32,10 @@
     pyc_abspath = os.path.join(root, pyc_relpath)
     try:
       py_compile.compile(abspath, cfile=pyc_abspath, dfile=relpath, doraise=True)
+      # Overwrite the timestamp in the .pyc file with 2000-01-01 so it's deterministic.
+      with open(pyc_abspath, 'r+b') as f:
+          f.seek(4)
+          f.write(b'\\x80Cm8')
       compiled.append(pyc_relpath)
     except py_compile.PyCompileError as e:
       errored[e.file] = e.msg
