--- coverage/python.py	2015-07-08 20:31:40.183290075 +0100
+++ coverage/python.py.new	2015-07-08 20:35:55.269446685 +0100
@@ -63,7 +63,7 @@
     an empty string if the file is empty.
 
     """
-    markers = ['.zip'+os.sep, '.egg'+os.sep]
+    markers = ['.zip'+os.sep, '.egg'+os.sep, '.pex'+os.sep]
     for marker in markers:
         if marker in filename:
             parts = filename.split(marker)
