--- pex/pex_builder.py	2017-03-30 10:11:00.140019000 +0100
+++ pex/pex_builder.py.new	2017-05-04 10:32:49.555016007 +0100
@@ -337,7 +337,7 @@
       self._chroot.touch(compiled, label='bytecode')
 
   def _prepare_manifest(self):
-    self._chroot.write(self._pex_info.dump().encode('utf-8'), PexInfo.PATH, label='manifest')
+    self._chroot.write(self._pex_info.dump(sort_keys=True).encode('utf-8'), PexInfo.PATH, label='manifest')
 
   def _prepare_main(self):
     self._chroot.write(self._preamble + b'\n' + BOOTSTRAP_ENVIRONMENT,
