--- pex/common.py	2015-06-28 13:08:26.627711606 +0100
+++ pex/common.py.new	2015-06-28 13:09:26.164961700 +0100
@@ -310,4 +310,6 @@
   def zip(self, filename, mode='wb'):
     with contextlib.closing(zipfile.ZipFile(filename, mode)) as zf:
       for f in sorted(self.files()):
-        zf.write(os.path.join(self.chroot, f), arcname=f, compress_type=zipfile.ZIP_DEFLATED)
+        filename = os.path.join(self.chroot, f)
+        compression = zipfile.ZIP_STORED if zipfile.is_zipfile(filename) else zipfile.ZIP_DEFLATED
+        zf.write(filename, arcname=f, compress_type=compression)
