--- pex/environment.py	2016-01-09 19:12:00.314371772 +0000
+++ pex/environment.py.new	2016-01-09 19:18:01.887465480 +0000
@@ -5,6 +5,7 @@

 import itertools
 import os
+import shutil
 import site
 import sys
 import uuid
@@ -34,6 +35,10 @@
       return pex
     explode_dir = os.path.join(pex_info.zip_unsafe_cache, pex_info.code_hash)
     TRACER.log('PEX is not zip safe, exploding to %s' % explode_dir)
+    # plz modifies the .pex file after pex runs, so do not reuse the cached version ever.
+    # plz test does its own incrementality anyway.
+    if os.path.exists(explode_dir):
+      shutil.rmtree(explode_dir)
     if not os.path.exists(explode_dir):
       explode_tmp = explode_dir + '.' + uuid.uuid4().hex
       with TRACER.timed('Unzipping %s' % pex):
