diff --git a/parser.go b/parser.go
index fd2fd5f..3af2af9 100644
--- a/parser.go
+++ b/parser.go
@@ -633,10 +633,6 @@ func (p *parseState) addArgs(args ...string) error {
 }
 
 func (p *Parser) parseNonOption(s *parseState) error {
-	if len(s.positional) > 0 {
-		return s.addArgs(s.arg)
-	}
-
 	if len(s.command.commands) > 0 && len(s.retargs) == 0 {
 		if cmd := s.lookup.commands[s.arg]; cmd != nil {
 			s.command.Active = cmd
@@ -649,6 +645,10 @@ func (p *Parser) parseNonOption(s *parseState) error {
 		}
 	}
 
+	if len(s.positional) > 0 {
+		return s.addArgs(s.arg)
+	}
+
 	if (p.Options & PassAfterNonOption) != None {
 		// If PassAfterNonOption is set then all remaining arguments
 		// are considered positional
