diff --git a/set.go b/set.go
index 6ca1051..43de5dc 100644
--- a/set.go
+++ b/set.go
@@ -201,6 +201,19 @@ func set(cfg interface{}, sect, sub, name string, blank bool, value string) erro
 	}
 	if vSect.Kind() == reflect.Map {
 		vst := vSect.Type()
+		if vst.Key().Kind() == reflect.String && vst.Elem().Kind() == reflect.String {
+			if vSect.IsNil() {
+				vSect.Set(reflect.MakeMap(vst))
+			}
+			if name != "" {
+				if sub != "" {
+					vSect.SetMapIndex(reflect.ValueOf(sub + " " + name), reflect.ValueOf(value))
+				} else {
+					vSect.SetMapIndex(reflect.ValueOf(name), reflect.ValueOf(value))
+				}
+			}
+			return nil
+		}
 		if vst.Key().Kind() != reflect.String ||
 			vst.Elem().Kind() != reflect.Ptr ||
 			vst.Elem().Elem().Kind() != reflect.Struct {
