diff --git a/glog.go b/glog.go
index 8c00e73..3b1d59c 100644
--- a/glog.go
+++ b/glog.go
@@ -209,6 +209,7 @@ func logf(depth int, severity logsink.Severity, verbose bool, stack stack, forma
 // ctxlogf writes a log message for a log function call (or log function wrapper)
 // at the given depth in the current goroutine's stack.
 func ctxlogf(ctx context.Context, depth int, severity logsink.Severity, verbose bool, stack stack, format string, args ...any) {
+	return
 	now := timeNow()
 	_, file, line, ok := runtime.Caller(depth + 1)
 	if !ok {
@@ -399,6 +400,7 @@ type Verbose bool
 // V is at most the value of -v, or of -vmodule for the source file containing the
 // call, the V call will log.
 func V(level Level) Verbose {
+	return Verbose(false)
 	return VDepth(1, level)
 }
 
