diff --git a/semver/semver.go b/semver/semver.go
index 110fc23..813af52 100644
--- a/semver.go
+++ b/semver.go
@@ -133,6 +133,11 @@ func (v *Version) UnmarshalJSON(data []byte) error {
 	return v.Set(string(data[1 : l-1]))
 }

+// UnmarshalTest implements the encoding.TextUnmarshaler interface
+func (v *Version) UnmarshalText(text []byte) error {
+	return v.Set(string(text))
+}
+
 // Compare tests if v is less than, equal to, or greater than versionB,
 // returning -1, 0, or +1 respectively.
 func (v Version) Compare(versionB Version) int {
